pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -239,6 +243,21 @@ def on_validation_end(self, trainer, pl_module) -> None:
             return
         self.save_checkpoint(trainer)
 
+    def on_train_end(self, trainer, *args, **kwargs) -> None:
+        """
+        checkpoints can be saved at the end of the trianing
+        """
+        if not self._trigger_on_train_end:
+            return
+        # as we advance one step at end of training, we use global_step - 1
+        # to avoid saving duplicates
+        trainer.global_step -= 1
+        if (not self._should_skip_saving_checkpoint(trainer) and trainer.checkpoint_connector.has_trained):
+            if self.save_last and self.verbose:
+                rank_zero_info("Saving last checkpoint...")
+            self.save_checkpoint(trainer, is_on_train_end=True)
+        trainer.global_step += 1
```

Comment:
@shuyingsunshine21 can this not directly call `self._save_last_checkpoint` ? I think the most common case will be for saving a `last.ckpt` file at the end of training. this way we don't thread through the `is_on_train_end` flag everywhere

@carmocca what do you think? would this go along with https://github.com/PyTorchLightning/pytorch-lightning/issues/6470 ?